$OpenBSD: patch-lib_awful_ewmh_lua_in,v 1.1 2016/03/01 11:42:24 dcoppa Exp $

commit 9d34f2dc030a2bb95ad1e3d9ffdd27b9da039419
Author: Uli Schlachter <psychon@znc.in>
Date:   Mon Feb 15 21:15:41 2016 +0100

Fix awful.ewmh to handle window gravities

Since commit b2aaefd095c964491f2c93a, we correctly handle window gravities when
the border width of a client changes. Since most windows out there have a
NorthWest gravity, this means that most windows do not have this problem.
However, e.g. mplayer uses gravity "Static" and this causes this issue (any
gravity other than NorthWest will do).

This affects the fullscreen handling in awful.ewmh. The code has to set the
border width before it changes a client's geometry so that the move when the
border width changes doesn't matter.

No new integration test for this since I didn't find anything usable with a
non-NorthWest gravity. A test would be easy to write, just test if `c.fullscreen
= true ; c.fullscreen = false` restores the previous window geometry.

Fixes: https://github.com/awesomeWM/awesome/issues/697
Signed-off-by: Uli Schlachter <psychon@znc.in>

--- lib/awful/ewmh.lua.in.orig	Sat Jan 30 14:55:18 2016
+++ lib/awful/ewmh.lua.in	Tue Mar  1 11:24:18 2016
@@ -66,11 +66,11 @@ local function fullscreen(window, set)
         store_geometry(window, "fullscreen")
         data[window].fullscreen.border_width = window.border_width
         local g = screen[window.screen].geometry
-        window:geometry(screen[window.screen].geometry)
         window.border_width = 0
+        window:geometry(screen[window.screen].geometry)
     elseif data[window] and data[window].fullscreen then
-        window:geometry(data[window].fullscreen)
         window.border_width = data[window].fullscreen.border_width
+        window:geometry(data[window].fullscreen)
     end
 end
 
